src/components/ZetkinCalendar/utils.ts

```diff
@@ -0,0 +1,33 @@
+export enum CALENDAR_RANGES {
+    MONTH = 'month',
+    WEEK = 'week',
+}
+
+interface FirstAndLastDayInView {
+    firstDayInView : Date;
+    lastDayInView : Date;
+}
+
+/**
+ * Get first and laste date in visible range for week or month calendar views
+*/
+export const getViewRange = (focusDate : Date, range : CALENDAR_RANGES) : FirstAndLastDayInView => {
+    if (range === CALENDAR_RANGES.MONTH) {
+        const year = focusDate.getFullYear();
+        const month = focusDate.getMonth();
+        const firstMonthDay = new Date(year, month, 1);
+        const firstDayInView = new Date(new Date(firstMonthDay).setDate(firstMonthDay.getDate() - (firstMonthDay.getDay() || 7) + 1 ));
+        const lastMonthDay = new Date(year, month + 1, 1);
+        const lastDayInView = new Date(lastMonthDay.getFullYear(), lastMonthDay.getMonth(), (8 - lastMonthDay.getDay()) % 7 + 1);
+
+        return { firstDayInView, lastDayInView };
+    }
+    else {
+        // ranges == CALENDAR_RANGES.WEEK
+        const dayIndexFromMonday = (focusDate.getDay() + 6) % 7;
+        const firstDayInView = new Date(focusDate.getFullYear(), focusDate.getMonth(), focusDate.getDate() - dayIndexFromMonday);
+        const lastDayInView = new Date(focusDate.getFullYear(), focusDate.getMonth(), focusDate.getDate() + (7-dayIndexFromMonday));
```

Comment:
Why not use the `getNewDateWithOffset` utility function available in `dateutils.ts`? and just do `lastDayInView = getNewDateWithOffset(firstDateInView,  7)`

Reply:
That's true! That would be cleaner. But I think it should be seven (not six) days because I believe the `lastDayInView` needs to be the midnight on the following Monday.

Reply:
I don't have `dateutils.ts` in this branch. Is it a recent merge?

Reply:
It's available on `main`. Are you comfortable merging `main` into your branch and pushing it? Otherwise lets leave it as it is for now.